### PR TITLE
Expanded report to support lineage data

### PR DESCRIPTION
**Pull request by IDoneShaveIt**
- Created new lineage API for the report
- Added lineage to the report data
- Added an API for returning the lineage as list of dags

### CODE REVIEW

File: monitor/dbt_project/macros/get_models_depends_on_nodes.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% macro get_models_depends_on_nodes() %}
+    {% set models_depends_on_nodes_query %}
+        with dbt_models as (
+            select * from {{ref('elementary', 'dbt_models')}}
```
Comment (elongl) [inline monitor/dbt_project/macros/get_models_depends_on_nodes.sql:4]:
```suggestion
            select * from {{ ref('elementary', 'dbt_models') }}
```

**Review by elongl — COMMENTED** [review]
Looks really good! Well written.

File: monitor/api/lineage/schema.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @validator("depends_on_nodes", pre=True, always=True)
+    def set_depends_on_nodes(cls, depends_on_nodes):
+        return depends_on_nodes or []
+
+
+class LineageSchema(BaseModel):
+    nodes: Optional[List[ModelUniqueIdType]] = None
+    edges: Optional[List[Tuple[ModelUniqueIdType, ModelUniqueIdType]]] = None
+
+    @validator("nodes", pre=True, always=True)
```
[the remark this answers is not in the document]
Comment (IDoneShaveIt) [reply monitor/api/lineage/schema.py:21]: The problem with default value as list is that the schema breaks when you provide none.
So if you want to be able to get nones (which is safer IMO) you need a validator for having a default list 🙂

[the remark this answers is not in the document]
Comment (elongl) [reply monitor/api/lineage/schema.py:21]: Why do you want to support providing `None`?  
Those are lists, they can be empty.

[the remark this answers is not in the document]
Comment (elongl) [reply monitor/api/lineage/schema.py:21]: I understand.

**Review by elongl — COMMENTED** [review]
LGTM!

**Pull request merged**